.github/workflows/check-merge-conflicts.yml

```diff
@@ -0,0 +1,16 @@
+name: "Check merge conflicts"
+on:
+  push:
```

Comment:
Why do we need `push`?

Reply:
There won't be an `github.event.pull_request` for `push` events!

Reply:
> Why do we need `push`?

This action runs on every push and checks all pending PR's for merge conflicts, see https://github.com/jxs/rust-libp2p/actions/runs/3158064326/jobs/5139673243

> There won't be an github.event.pull_request for push events!

Yeah found out [here](https://github.com/jxs/rust-libp2p/pull/1#issuecomment-1263397962) :sweat_smile:

Reply:
But why do we need to run it on `push`?

Shouldn't `synchronize` happen every time HEAD or base changes? I guess one is always subscribed to a PR that you created so we don't necessarily need to tag the author, I'd be kinda nice though.

Reply:
This is still an open point for me @jxs.

Reply:
sorry Thomas missed this. 

>Shouldn't synchronize happen every time HEAD or base changes? 

yeah, but base changing seems to be different from "pushing commits to base", as per [doc](https://docs.github.com/en/github-ae@latest/developers/webhooks-and-events/webhooks/webhook-events-and-payloads#webhook-payload-object-28):
>synchronize: Triggered when a pull request's head branch is updated. For example, when the head branch is updated from the base branch, when new commits are pushed to the head branch, or when the base branch is changed. 

I assume base branch is changed, means changing the base branch on the PR. I tested it [here](https://github.com/jxs/rust-libp2p/pull/2)

Reply:
Oh damn that is annoying! Ah well, can't help it then.

Reply:
Hang on, how does this even work then? If this triggers on push to `master` f.e. how does the action know which PR to comment on?

Reply:
yeah, it seems to check out all PR's and compare to master to check for merge conflicts, while searching to confirm it found [this](https://github.com/eps1lon/actions-label-merge-conflict/issues/84) which might byte us, as we have time limits on the CI right?

Reply:
Hmm, that is a bit of a concern, we usually have lots of PRs open. @elenaf9 @mxinden @jxs Could I get your opinion on #3010? Happy for you to also suggest other tools but mergify could give us quite a bit of automation like [closing stale PRs](https://docs.mergify.com/configuration/#id7).

Reply:
FYI, you already have a workflow for closing up stale PRs - https://github.com/libp2p/rust-libp2p/blob/master/.github/workflows/stale.yml It's currently closing only those that need author input and have no activity.

Reply:
I also checked https://github.com/eps1lon/actions-label-merge-conflict/blob/main/sources/main.ts. It doesn't check out each PR branch. It uses the API to check on the mergeability status of all PRs. So the approach they're taking seems correct and it should work just fine for this repo. We can investigate further and contribute back to the project if it turns out it doesn't.

Reply:
ah nice then, thanks @galargh!

Reply:
> FYI, you already have a workflow for closing up stale PRs - [`master`/.github/workflows/stale.yml](https://github.com/libp2p/rust-libp2p/blob/master/.github/workflows/stale.yml?rgh-link-date=2022-10-13T08%3A12%3A18Z) It's currently closing only those that need author input and have no activity.

Interesting! haha

I wasn't aware at all and should probably go buy some glasses as I've completely overlooked that one all this time when I was working on the workflows!

We also don't really seem to use it / the label. We should probably keep that one then and just remove the label requirement so we don't have to explicitly tag PRs for it to run.